### ISSUE TITLE
Cluster Pool routees created on unreachable status instead of on member removed

### ISSUE THREAD

**Issue by srinivasrb** [akkadotnet/akka.net#4504]
When using a cluster pool router, when a node on which routees are deployed becomes unreachable, the router recreates the impacted routees on other nodes. However, since the unreachability of the node may be transient, the routees on that node may still be executing, thus resulting in 'extra' working routees.

Instead, the router should consider adding new routees only if the node is removed from the cluster, and not when it is (possibly temporarily) unreachable.

There is the case when the node is never going to be reachable due to a CLR crash, so in this implementation, new routees will not get recreated, but the consideration is that there may be a split-brain resolver / auto-down-after configured, so eventually the node will get marked as removed if the user configures it thus.

The changes will be in `Akka.Cluster.Routing.ClusterRouterActor OnReceive`.

Please advise if there was other rationale for the current implementation.

Akka version 1.4.8
.Net version 3.1
Platform: Windows

**Comment by ismaelhamed** [comment]
>All routers can be made aware of member nodes in the cluster, i.e. deploying new routees or looking up routees on nodes in the cluster. When a node becomes unreachable or leaves the cluster the routees of that node are automatically unregistered from the router. When new nodes join the cluster, additional routees are added to the router, according to the configuration. Routees are also added when a node becomes reachable again, after having been unreachable.

So new routees should only be created when new nodes join the cluster, or existing unreachable nodes become reachable again. If that is not the case, this is worth looking into.

**Comment by srinivasrb** [comment]
The behavior is presently this:
- Consider 2 worker nodes, and a pool size of 10 - this creates 5 routees on each node
- Now, worker node 1 becomes temporarily unreachable. As soon as the unreachable status is detected, the router assumes that 5 routees have gone away, and creates 5 new routees on node 2.
- At this point, there are actually 15 routees performing work, which is different from the expected pool size of 10.

(Later, when worker node 1 becomes reachable, I think the pool size is readjusted (haven't checked this) but if it is, then it has the side-effect of potentially stopping working routees)

The issue is that since unreachability is potentially transient, it seems incorrect to assume that the routees on the unreachable node have completely stopped.

**Comment by IgorFedchenko** [comment]
@srinivasrb Isn't that an eventual delivery problem? I mean, in your example, after node 1 becomes unreachable, there are two options:

1. Assume that the node is still running, and routees are still doing their jobs well. 
In the best case, node will be back, and we have 10 routees all the time. 
In the worse case, we have only 5 routees on node 2 doing the job, and waiting for timeout for node 1 to be considered as dead and removed from cluster.

2. Assume that the node 1 is not working anymore, or has some issues, and add 5 more healthy routees on node 2 immediately. 
In the best case, routees on node 1 were not doing their jobs (could not connect to database/other nodes or just were shutdown), and we keep healthy routees number matching the requirement of 10.
In the worse case, routees on node 1 were still doing their jobs, and during the transient unreachability we had 15 routees instead of 10 required.

Currently, second option is implemented, and I guess you are suggesting to go with first option?
If so, I am not sure that this is a reasonable default, because usually you will prefer having more routees then required to handle the workload, then having less workers until the node becomes reachable or removed from cluster.

Basically, first option assumes that workers are healthy until they are completely shutdown and this is confirmed by timeout. Second option assumes that if workers are not reachable, they are not healthy (may have connectivity issues with database, etc), and we need to replace them as fast as possible to keep required number of healthy workers. 

Both options do have their downsides, and in both cases we may have healthy routees number less or greater then 10. Probably, would be nice to have a configuration switch to be able to go with first option, if we want routees number to be "less then or equal to" 10 instead of "greater then or equal to". But that would be more like feature request, not a change to the current behavior.

What do you think?
